src/gtk/window.cpp

```diff
@@ -2472,6 +2472,16 @@ bool wxGetKeyState(wxKeyCode WXUNUSED(key))
 }
 #endif // __WINDOWS__
 
+// Under Unix this is implemented using X11 functions in utilsx11.cpp but we
+// need to have this function under Wayland too, so provide at least a stub.
+#ifdef GDK_WINDOWING_WAYLAND
+bool wxGetKeyState(wxKeyCode WXUNUSED(key))
+{
+    wxFAIL_MSG(wxS("Not implemented under Wayland"));
+    return false;
+}
+#endif // __WAYLAND__
+
```

Comment:
I think this change shouldn't be necessary (at least with recent changes in master).  wxGetKeyState() should work on Wayland.

Reply:
The recent changes in master assume X11 is enabled too.
This is only true when using xwayland (X11 + wayland).
If the system support Wayland, but not X11 support (this is the case of GPUs in latest NXP iMX8 SoC families), the file
   src/unix/utilsx11.cpp
should not be compiled at all since it relies on X11 APIs.
In this corner case, the function wxGetKeyState is not defined.

Reply:
In the case of Wayland, `wxGetKeyStateGTK()` should be used, which should work on Wayland.  If that isn't the case, then we should fix that.

Reply:
My problem is that Wayland does not require X11 and I'm actually building a wayland based Yocto distro without X11.
Assuming the wxGetKeyStateGTK should be used in any case, probably it should be moved out from file
   src/unix/utilsx11.cpp
I would expect it should be placed in file
   src/gtk/window.cpp
to make it visible even if X11 is not enabled.
Does it sound reasonable?

Reply:
I understand completely what you're doing - no need to explain that.

I don't think there's any need to move anything - we can just fix src/unix/utilsx11.cpp if necessary so that all the X11 stuff is guarded appropriately.  What is the compile error you get with that file when using Wayland?

Reply:
The #include here
   https://github.com/wxWidgets/wxWidgets/blob/master/src/unix/utilsx11.cpp#L16
cause an error here
   https://github.com/wxWidgets/wxWidgets/blob/master/include/wx/unix/utilsx11.h#L18
Which is the expected way to wrap it?

Reply:
Would "#ifdef __WXX11__" be reasonable?

Reply:
No, it's more complicated than that unfortunately.  That Xlib header needs to be included for _WXMOTIF__ __WXGTK__ or __WXX11__, and when __WXGTK__, only when GDK_WINDOWING_X11 is defined.

Reply:
I suppose you mean something like
#if defined(__WXX11__) || (defined(__WXGTK__) && defined(GDK_WINDOWING_X11)) || defined(__WXMOTIF__)
...
#endif
Is this correct?

Reply:
Yes, exactly.